pkg/apis/navigator/validation/elasticsearch.go

```diff
@@ -0,0 +1,87 @@
+package validation
+
+import (
+	apimachineryvalidation "k8s.io/apimachinery/pkg/api/validation"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+
+	"github.com/jetstack/navigator/pkg/apis/navigator"
+)
+
+func ValidateImageSpec(img *navigator.ImageSpec, fldPath *field.Path) field.ErrorList {
+	el := field.ErrorList{}
+	if img.Tag == "" {
+		el = append(el, field.Required(fldPath.Child("tag"), ""))
+	}
+	if img.Repository == "" {
+		el = append(el, field.Required(fldPath.Child("repository"), ""))
+	}
+	return el
+}
+
+var supportedESClusterRoles = []string{
+	string(navigator.ElasticsearchRoleData),
+	string(navigator.ElasticsearchRoleIngest),
+	string(navigator.ElasticsearchRoleMaster),
+}
+
+func ValidateElasticsearchClusterRole(r navigator.ElasticsearchClusterRole, fldPath *field.Path) field.ErrorList {
+	el := field.ErrorList{}
+	switch r {
+	case navigator.ElasticsearchRoleData:
+	case navigator.ElasticsearchRoleIngest:
+	case navigator.ElasticsearchRoleMaster:
+	default:
+		el = append(el, field.NotSupported(fldPath, r, supportedESClusterRoles))
+	}
+	return el
+}
+
+func ValidatElasticsearchClusterNodePool(np *navigator.ElasticsearchClusterNodePool, fldPath *field.Path) field.ErrorList {
+	el := ValidateDNS1123Subdomain(np.Name, fldPath.Child("name"))
+	el = ValidateElasticsearchPersistence(&np.Persistence, fldPath.Child("persistence"))
```

Comment:
This clobbers any errors returned from `ValidateDNS1123Subdomain`.

Reply:
:+1: Ack